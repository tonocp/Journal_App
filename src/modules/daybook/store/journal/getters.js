// export const myGetter = ( state ) => {
// return state...
// };

export const getEntriesByTerm = (state) => {
  return state.entries;
};

export const getEntryById = (/* state */) => {};
