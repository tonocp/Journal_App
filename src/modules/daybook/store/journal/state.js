export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toLocaleDateString(),
      text: "Sit excepteur id consectetur dolore amet dolore qui laboris eu nulla irure. Esse amet pariatur enim veniam proident ad proident. Adipisicing laborum laboris ea elit. Proident duis nulla officia esse culpa ullamco aute anim ut duis tempor.",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toLocaleDateString(),
      text: "Culpa adipisicing amet enim anim cupidatat qui mollit nisi labore culpa tempor adipisicing ut est. Consectetur eu ad non adipisicing aliqua aliqua aliquip minim consequat officia sunt cupidatat veniam esse. Reprehenderit fugiat minim consequat adipisicing irure. Ea eiusmod duis incididunt non eiusmod laboris id qui deserunt velit pariatur velit. Commodo aliqua in id incididunt et enim ea dolor cillum sunt elit id pariatur eiusmod. Ipsum laborum eiusmod incididunt reprehenderit deserunt quis in irure dolor nisi laboris culpa ad esse.",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toLocaleDateString(),
      text: "Anim amet labore labore aliqua commodo irure duis veniam ipsum Lorem est est sit. In culpa quis exercitation id proident. Ea irure occaecat labore consequat nulla irure commodo eu labore quis eu dolor Lorem sint. Deserunt deserunt ad pariatur laboris enim aliqua.",
      picture: null,
    },
  ],
});
