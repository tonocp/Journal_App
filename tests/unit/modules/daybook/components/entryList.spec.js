import { createStore } from "vuex";
import { shallowMount } from "@vue/test-utils";

import journal from "@/modules/daybook/store/journal";
import EntryList from "@/modules/daybook/components/EntryList.vue";
import { journalState } from "../../../mock-data/test-journal-state";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

describe("Pruebas en el EntryList", () => {
  const store = createVuexStore(journalState);
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("Debe de llamar el getEntriesByTerm sin argumento y mostrar 2 entradas", () => {
    expect(wrapper.findAll("entry-stub").length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe de llamar el getEntriesByTerm y filtrar las entradas", async () => {
    const input = wrapper.find("input");
    await input.setValue("Holi");

    expect(wrapper.findAll("entry-stub").length).toBe(1);
  });

  test("El botÃ³n de nueva entrada debe redireccionar a /new", () => {
    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "entry", params: { id: "new" } });
  });
});
