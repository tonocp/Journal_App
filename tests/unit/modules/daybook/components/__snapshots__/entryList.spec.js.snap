// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryList Debe de llamar el getEntriesByTerm sin argumento y mostrar 2 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="d-flex flex-column mt-2"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
